sscanss/app/widgets/graphics.py

```diff
@@ -672,23 +1031,22 @@ def size(self, value):
         self.x, self.y = value
 
     def render(self, painter, rect):
-        scene_rect = rect.toRect()
-        left = scene_rect.left()
-        top = scene_rect.top()
-        right = scene_rect.right() + 2 * self.x
-        bottom = scene_rect.bottom() + 2 * self.y
+        left = rect.left()
+        top = rect.top()
+        right = rect.right() + 2 * self.x
+        bottom = rect.bottom() + 2 * self.y
 
         left = left - left % self.x
         top = top - top % self.y
 
-        x_offsets = np.array(range(left, right, self.x))
-        y_offsets = np.array(range(top, bottom, self.y))
+        x_offsets = np.arange(left, right, self.x)
+        y_offsets = np.arange(top, bottom, self.y)
```

Comment:
I don't think there is any need for these to be numpy arrays, so could just use the `range()`.

Personally I'd just do it inline with the for: `for x in range(left, right, self.x)`

Same for polar grid below

Reply:
Will move to inline